transmogrifier/sources/marc.py

```diff
@@ -1,8 +1,200 @@
 import logging
+
+from bs4 import Tag
+
+import transmogrifier.models as timdex
 from transmogrifier.sources.transformer import Transformer
 
 logger = logging.getLogger(__name__)
 
 
 class Marc(Transformer):
     """Marc transformer."""
+
+    def get_optional_fields(self, xml: Tag) -> dict:
+        """
+        Retrieve optional TIMDEX fields from a MARC XML record.
+
+        Overrides metaclass get_optional_fields() method.
+
+        Args:
+            xml: A BeautifulSoup Tag representing a single MARC XML record.
+        """
+        fields: dict = {}
+
+        # alternate_titles
+        alternate_title_marc_data = {
+            "130": ("adfghklmnoprst", "Main Entry - Uniform Title"),
+            "240": ("adfghklmnoprs", "Uniform Title"),
+            "246": ("abfghinp", "Varying Form of Title"),
+            "730": ("adfghiklmnoprst", "Added Entry - Uniform Title"),
+            "740": ("anp", "Added Entry - Uncontrolled Related/Analytical Title"),
+        }
+        for marc_tag, marc_field_data in alternate_title_marc_data.items():
+            for datafield in xml.find_all("datafield", tag=marc_tag):
+                if alternate_title_value := (
+                    self.create_subfield_value_string_from_datafield(
+                        datafield,
+                        marc_field_data[0],
+                        " ",
+                    )
+                ):
+                    fields.setdefault("alternate_titles", []).append(
+                        timdex.AlternateTitle(
+                            value=alternate_title_value,
+                            kind=marc_field_data[1],
+                        )
+                    )
+
+        # call_numbers
+        call_numbers_marc_data = {
+            "050": ("a", ""),
+            "082": ("a", ""),
+        }
+        for marc_tag, marc_field_data in call_numbers_marc_data.items():
+            for datafield in xml.find_all("datafield", tag=marc_tag):
+                if call_number_value := self.create_subfield_value_string_from_datafield(
+                    datafield,
+                    marc_field_data[0],
+                ):
+                    fields.setdefault("call_numbers", []).append(call_number_value)
+
+        # citation
+
+        # content_type
+
+        # contents
+
+        # contributors
+
+        # dates
+
+        # edition
+
+        # file_formats
+
+        # format
+
+        # funding_information
+
+        # holdings
+
+        # identifiers
+
+        # languages
+
+        # links
+
+        # literary_form
+        # Literary form is applicable to Book (BK) material configurations and indicated
+        # by leader "Type of Record" position = "Language Material" or "Manuscript
+        # language material" and "Bibliographic level" position =
+        # "Monographic component part," "Collection," "Subunit," or "Monograph/Item."
+        if leader := xml.find("leader", string=True):
```

Comment:
If `leader` and `fixed_length_data` end up being called multiple times, I will move them to the top as we did with a few `Ead` variables